CASMTRIAGE-7199 : supply options -sv and/or -bpcd or -rv during IUF deliver-product stage to install SLURM or PBS (#5308)
Cray-HPE/docs-csm: operations/iuf/workflows/product_delivery.md

Refer to that table and any corresponding product documents before continuing to the next step.

1. Invoke `iuf run` with activity identifier `${ACTIVITY_NAME}` and use `-r` to execute the [`deliver-product`](../stages/deliver_product.md) stage. Perform the upgrade using product content found in `${MEDIA_DIR}`.
   Additional arguments are available to control the behavior of the `deliver-product` stage (for example, `-rv`). See the [`deliver-product` stage documentation](../stages/deliver_product.md)
   for details and adjust the example below if necessary.

>      **`NOTE`** Enable SLURM or PBS Pro Products when installing USS 1.1 or higher. Select which workload manager to use on the system (either SLURM or PBS). For more information, see the `deliver-product` stage

Reviewer: ```suggestion
     **`NOTE`** When installing USS 1.1 or higher, select either SLURM or PBS Pro Products to use on the system before running this stage. For more information, see the `deliver-product` stage
```

1. Invoke `iuf run` with activity identifier `${ACTIVITY_NAME}` and use `-r` to execute the [`deliver-product`](../stages/deliver_product.md) stage. Perform the upgrade using product content found in `${MEDIA_DIR}`.

    1. Option 1: Install or upgrade without SLURM or PBS Products

       (`ncn-m001#`) Execute the `deliver-product` stage.

       ```bash
       iuf -a ${ACTIVITY_NAME} -m "${MEDIA_DIR}" run -r deliver-product
       ```

>     1. Option 2: Install or upgrade along with SLURM or PBS Products

Reviewer 2: SLURM and PBS Pro are technically included in the USS product starting in version 1.1.   So we might want to make that clearer in the description.   

`Option 2:  Enable SLURM or PBS Pro Products when installing USS 1.1 or higher`

Reviewer: You make a good point, and admin actually need to select which WLM software to use by updating the site variables before running this stage.

       **`NOTE`** This subsection is mandatory only while installing SLURM or PBS Products. Additional arguments are available to control the behavior of the `deliver-product` stage, for example `-rv`. See the [`deliver-product`
       stage documentation](../stages/deliver_product.md) for details and adjust the examples below if necessary.

       1. Use site variables from the `site_vars.yaml` file found in `${ADMIN_DIR}` and recipe variables from the `product_vars.yaml` file found in `${ADMIN_DIR}`.

           (`ncn-m001#`) Execute the `deliver-product` stage.

           ```bash
           iuf -a ${ACTIVITY_NAME} -m "${MEDIA_DIR}" run --site-vars "${ADMIN_DIR}/site_vars.yaml" -bpcd "${ADMIN_DIR}" -r deliver-product

>            ```

Reviewer: 1. Invoke `iuf run` with activity identifier `${ACTIVITY_NAME}` and use `-r` to execute the [`deliver-product`](../stages/deliver_product.md) stage. Perform the upgrade using product content found in `${MEDIA_DIR}`.

   - Option 1: Install or upgrade without SLURM or PBS Products

      (`ncn-m001#`) Execute the `deliver-product` stage.

      ```bash
      iuf -a ${ACTIVITY_NAME} -m "${MEDIA_DIR}" run -r deliver-product
      ```

   - Option 2: Enable a workload manager with USS
 
     **`NOTE`** This subsection is mandatory only while installing a workload manager with USS 1.1 or higher. Additional arguments are available to control the behavior of the `deliver-product` stage (for example, `-rv`). See the [`deliver-product` stage documentation](../stages/deliver_product.md) for details and adjust the example below if necessary.

     1. Select which workload manager to use on the system (either SLURM or PBS). For more information, see the `deliver-product` stage details in the "Install and Upgrade Framework" section of the [*HPE Cray Supercomputing User Services Software Administration Guide: CSM on HPE Cray Supercomputing EX Systems (S-8063)*](https://www.hpe.com/support/COS-S-8063-2407).

     1. (`ncn-m001#`) Execute the `deliver-product` stage. Use site variables from the `site_vars.yaml` file found in `${ADMIN_DIR}` and recipe variables from the `product_vars.yaml` file found in `${ADMIN_DIR}`.

        ```bash
        iuf -a ${ACTIVITY_NAME} -m "${MEDIA_DIR}" run --site-vars \
        "${ADMIN_DIR}/site_vars.yaml" -bpcd "${ADMIN_DIR}" -r deliver-product
        ```

Reviewer: I rewrote this section to add more details.

Reviewer: In order to enable a workload manager, admin also need to include the site variables file with the `deploy-product` stage. I think we need to include these changes in the [Deploy product workflow](https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/iuf/workflows/deploy_product.md) as well.

We should add similar options to step 2 of the [Execute the IUF `deploy-product` stage](https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/iuf/workflows/deploy_product.md#1-execute-the-iuf-deploy-product-stage). Note that the `--site-vars` argument needs to be added, but the WLM does not need to be selected in the site variable file again (since it was already selected here for the `deliver-product` stage).
